File: the-drop-shop-service/src/main/java/com/example/thedropshopservice/entities/Product.java
package com.example.thedropshopservice.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@Document("products")
public class Product {
    @Id
    private String productId;
    private String title;
    private String description;
    private Double price;
    private String brand;
    private List<ObjectId> imageIds;
    private List<Size> sizes;
}

File: the-drop-shop/src/types/product.ts
export interface Product {
    productId: string;
    title: string;
    description: string;
    brand: string;
    price: number;
    images: Image[];
    sizes: Size[];
}

export interface Image {
    imageId: string;
    name: string;
    data: {
        //returned as base64
        data: string;
    };
}

export interface Size {
    sizeId: string;
    size: number;
    stock: number;
}